src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+    let workSep = separator;
+
+    if (workSep === null) {
+      workSep = 'null';
+    }
+
+    if (this.length) {
+      for (let i = 0; i < this.length; i++) {
+        if (this[i] === null || this[i] === undefined) {
+          this[i] = '';
```

Comment:
I don't think, that changing the source array in this method is a good idea. Try to avoid it.

Reply:
he don't modify the original array

Reply:
You do modify the original array. Original join method doesn't do it.